### ISSUE TITLE
Export result as HTML with autoscroll enabled?

### ISSUE THREAD

**Issue by bagage** [pgaufillet/meos#36]
I'm trying to export results to HTML with autoscrolling enabled and then display this .html file on another device to let everyone see results. 
I started Print / Export results service and do an Export with structured export format and HTML with AutoRefresh tickboxes enabled, but the HTML file won't scroll. Am I missing something?

**Comment by bagage** [comment]
edit: ok I actually misunderstood that Autorefresh would reload the HTML page and not do the scrolling.

Still, if someone knows a solution to achieve that directly from meos (eg. without having to invoke an after export script which would be my last resort), I'd gladly take any hint :).

**Comment by jmonclard** [comment]
Are you expecting an after competition display or something updated during the competition ?

IMHO, I think there isn't currently any auto-scrolling html file generation. The result display that scrolls by itself is internal to MeOS.
HTML generation is mainly for results diffusion of final results on websites.
If you are interested in real time updates, services (online results) is probably the way to go.

Adding scrolling to exported HTML code could be added embedding some javascript in the generated file.  Probably a nice to have feature for training/(very) small competitions to add one or two more screens. Will require some MeOS code change though.
For medium to large competitions you could have a look to my MopScreens set of php codes that work closely with MeOS and just handle real time results display with bells and whistles (including scrolling) but requires to install a web page server.

**Comment by bagage** [comment]
I'd like to use autoscrolling for real time update during the competition indeed. For now, I quickly wrote a dirty python script injecting the following autoscrolling code to the end of generated `.html` file:

```js
<script>
  var dir = 1
  var extremity_wait_sec = 2
  var scroll_sec = .05
  var scroll_amount = 2
  function scrollMore() {
    var current = window.pageYOffset
    window.scrollBy(0, scroll_amount * dir)
    if (dir != 0 && Math.abs(current - window.pageYOffset) < scroll_amount / 2.0) {
      var nextDir = dir * -1
      dir = 0
      setTimeout(function() { dir = nextDir }, 1000 * extremity_wait_sec)
    }
  }
  setInterval(scrollMore, 1000 * scroll_sec)
</script>
```

These files will be served via a simple python HTTP server `python -m SimpleHTTPServer`. Hopefully this should do the trick for this time -- I'd take a look at MopScreens if competitions get bigger, thanks @jmonclard!

**Comment by jmonclard** [comment]
Thanks for sharing your code @bagage.
I have some modifications to do in MeOS on the online export side but unfortunately will not have time to work on them during the month to come. If nobody else add the script during the while I will add that on my todo list as the changes look simple enough and the feature nice to have.

**Comment by jmonclard** [comment]
Job done in a dedicated branch and pushed in my GitHub (in meos fork).
A pull request has been done to the main project manager, so the changes may be in a next release.

**Comment by bagage** [comment]
I slightly updated the script to fix some corner edge issues (save current direction in storage to properly continue scrolling on page auto reload mainly):

```
  <script>
    var dir = sessionStorage.getItem('direction') || 1
    var extremity_wait_sec = 2
    var scroll_sec = .05
    var scroll_amount = 2
    function scrollMore() {

        window.scrollBy(0, scroll_amount * dir)

        var current = window.pageYOffset + window.innerHeight
        var body = document.body, html = document.documentElement;
        var height = Math.max( body.scrollHeight, body.offsetHeight,
                                html.clientHeight, html.scrollHeight, html.offsetHeight );

      if ((dir > 0 && current > .95 * height) || (dir < 0 && current <= window.innerHeight)) {
        var nextDir = dir * -1
        console.log("reached " + (dir > 0 ? "bottom" : "top") + " at ", new Date())
        dir = 0
        setTimeout(function() {
            dir = nextDir
            sessionStorage.setItem('direction', dir)
        }, 1000 * extremity_wait_sec)
      }
    }
    setInterval(scrollMore, 1000 * scroll_sec)
  </script>
```

**Comment by jmonclard** [comment]
I made the changes, but I slightly modified your script as the bottom of the page was not displayed: scroll stopped and reversed about half a page before the end of the file for my test case.
I replaced:
`if ((dir > 0 && current > .95 * height) ...`
with: 
`if ((dir > 0 && current >= height)... `

Well, if there is a risk with the equality, replacing 0.95 with 0.9999 should do the trick.

BTW, as I'm new to Git, I messed up my repository a bit in the process. The branch should be fine now. If you need an exe file to do your own tests and don't want to rebuild MeOS let me know.

**Comment by bagage** [comment]
Yeah, I saw that bug today while running live -- apart of that the process worked great! I'll try your version tomorrow, I also added scroll position resume, which seems needed on firefox@iPad at least:

```diff
    var dir = sessionStorage.getItem('direction') || 1
+    window.scrollBy(0, sessionStorage.getItem('scroll_position'))
    var extremity_wait_sec = 2
    var scroll_sec = .0
    var scroll_amount = 2
    function scrollMore() {

        window.scrollBy(0, scroll_amount * dir)
+        sessionStorage.setItem('scroll_position', window.pageYOffset)
        var current = window.pageYOffset + window.innerHeight
        ...

```
